.changeset/full-places-know.md

```diff
@@ -0,0 +1,38 @@
+---
+"fluid-framework": minor
+"@fluidframework/tree": minor
+---
+---
+"section": tree
+---
+
+Fix typing bug in `adaptEnum` and `enumFromStrings`
+
+When using the return value from [`adaptEnum`](https://fluidframework.com/docs/api/v2/tree#adaptenum-function) as a function, passing in a value who's type is a union no longer produced an incorrectly typed return value. This has been fixed.
+
+Additionally [`enumFromStrings`](https://fluidframework.com/docs/api/v2/tree#enumfromstrings-function) has improved the typing of its schema, ensuring the returned object's members have sufficiently specific types.
+Part of this improvement was fixing the `.schema` property to be a tuple over each of the schema where it was previously a tuple of a single combined schema due to a bug.
+
+One side-effect of these fixes is that narrowing of the `value` field of a node typed from the `.schema` behaves slightly different, such the node type is now a union instead of it being a single type with a `.value` that is a union.
+This means the `.value` property needs to be read into its own variable to be able to be narrowed (for example in a switch statement).
+
+```typescript
+const Mode = enumFromStrings(schema, ["Fun", "Bonus"]);
+type Mode = TreeNodeFromImplicitAllowedTypes<typeof Mode.schema>;
```

Comment:
(Not for this PR) The more I see `TreeNodeFromImplicitAllowedTypes` (and similarly named types) appear in public-facing scenarios, the more I think we may want to consider renaming. I don't think the current name will make a lot of sense to the average user, and it's even less discoverable.

Reply:
We can't rename it, but we can add a differently named alternative. Insertable and InsertableField are candidates for that. Both do work here, but they are alpha APIs which could get stabilized independently from this code.

Reply:
I don't know what our process is for scheduling work for promoting things from alpha, but if you want those API stabilized, subjecting them to APi review might be the next step?

Reply:
> We can't rename it, but we can add a differently named alternative. Insertable and InsertableField are candidates for that. Both do work here, but they are alpha APIs which could get stabilized independently from this code.

Yeah, my assumption was we would need to do it via an aliased export or something.